mlflow/projects/_project_spec.py

```diff
@@ -83,6 +83,18 @@ def _validate_parameters(self, user_parameters):
                 "No value given for missing parameters: %s" %
                 ", ".join(["'%s'" % name for name in missing_params]))
 
+    def consolidate_parameters(self, user_parameters):
+        final_params = {}
```

Comment:
Nit: Can we just update the user params? (If we want to preserve user params vs default we can create a clone._ I.e. replace the method with:

```
for name, param_obj in self.parameters.items():
     final_params = user_params.copy() # clone so that we preserve original user params
     if name not in user_parameters:
         final_params[name] = param_obj.default
```

Reply:
I can do that.